cmd/tether/attach.go

```diff
@@ -140,78 +171,121 @@ func (t *attachServerSSH) start() error {
 	}
 	t.sshConfig.AddHostKey(pkey)
 
-	t.enabled = true
+	t.Enabled()
 	go t.run()
 
 	return nil
 }
 
 // stop is not thread safe with start
-func (t *attachServerSSH) stop() {
+func (t *attachServerSSH) stop() error {
 	defer trace.End(trace.Begin("stop attach server"))
 
-	if t == nil || !t.enabled {
-		return
+	if t == nil {
+		err := fmt.Errorf("attach server is not configured")
+		log.Error(err)
+		return err
+	}
+
+	if !t.IsEnabled() {
+		err := fmt.Errorf("attach server is not enabled")
+		log.Error(err)
+		return err
 	}
 
-	t.enabled = false
-	conn := t.conn
-	t.conn = nil
+	log.Debugf("Setting enabled to false")
+	t.Disabled()
 
-	if conn != nil {
-		(*conn).Close()
+	log.Debugf("Canceling the context")
+	t.cancel()
+
+	log.Debugf("Acquiring the connection lock")
+	t.conn.Lock()
+	if t.conn.conn != nil {
```

Comment:
@fdawg4l there is an error path starting at line 250 that I think would result in this.